api/internal/localizer/locloader.go

```diff
@@ -0,0 +1,382 @@
+func defaultNewDir(targetLdr ifc.Loader, spec *git.RepoSpec) string {
+	targetDir := filepath.Base(targetLdr.Root())
+	if repo, isRemote := targetLdr.Repo(); isRemote {
+		// kustomize doesn't download repo into repo-named folder
+		// must find repo folder name from url
+		if repo == targetLdr.Root() {
+			targetDir = urlBase(spec.OrgRepo)
+		}
+		return strings.Join(
+			[]string{dstPrefix, targetDir, strings.ReplaceAll(spec.Ref, "/", "-")},
+			"-")
+	}
+	// special case for local target directory since destination directory cannot have "/" in name
+	if targetDir == string(filepath.Separator) {
+		return dstPrefix
+	}
+	return strings.Join([]string{dstPrefix, targetDir}, "-")
+}
+
+// validateNewDir returns the localize destination directory or error. Note that spec is nil if targetLdr is at local
+// target.
+func validateNewDir(newDirArg string, targetLdr ifc.Loader, spec *git.RepoSpec, fSys filesys.FileSystem) (filesys.ConfirmedDir, error) {
+	var newDirPath string
+	if newDirArg == "" {
+		newDirPath = defaultNewDir(targetLdr, spec)
+	} else {
+		newDirPath = newDirArg
+	}
+	if fSys.Exists(newDirPath) {
+		return "", errors.WrapPrefixf(
+			errors.Errorf("localize destination already exists"),
+			"invalid localize destination path '%s'", newDirPath)
+	}
+	// destination directory must sit in existing directory
+	if err := fSys.Mkdir(newDirPath); err != nil {
+		return "", errors.WrapPrefixf(err, "unable to create localize destination directory")
+	}
+	newDir, err := filesys.ConfirmDir(fSys, newDirPath)
+	if err != nil {
+		if err := fSys.RemoveAll(newDir.String()); err != nil {
+			log.Printf("%s", errors.WrapPrefixf(err, errCleanDst).Error())
+		}
+		return "", errors.WrapPrefixf(err, "unable to establish localize destination")
+	}
+
+	return newDir, nil
+}
+
+// pathSuffix returns path relative to prefix, where both are cleaned absolute paths
+// and prefix is a directory known to contain path
+func pathSuffix(prefix string, path string) string {
+	if prefix == string(filepath.Separator) || prefix == path {
+		return path[len(prefix):]
+	}
+	return path[len(prefix)+1:]
+}
+
+// Root returns the path to the current target directory
+func (ll *LocLoader) Root() string {
+	return ll.ldr.Root()
+}
+
+// Dst returns the path to the directory in destination that corresponds to Root
+func (ll *LocLoader) Dst() string {
+	return ll.dst
+}
+
+func toFSPath(urlPath string) string {
+	return filepath.Join(strings.Split(urlPath, "/")...)
+}
+
+func locFilePath(u *url.URL) string {
+	// raw github urls are the only type of file urls kustomize accepts;
+	// path consists of org, repo, version, path in repo
+	return filepath.Join(localizeDir, u.Hostname(), toFSPath(u.Path))
+}
+
+// Load returns the contents of path and its corresponding localize destination path information
+// if path is a file that can be localized. Otherwise, Load returns an error.
+func (ll *LocLoader) Load(path string) ([]byte, *LocPath, error) {
```

Comment:
nit: LocLoader and its methods should be grouped together. It may be easier to do that if `locloader.go` _just_ contains LocLoader & methods, while the other helper functions or business logic is in a separate file